### PR TITLE
Tools: added mav-firmware-version-str to manifest

### PR DESCRIPTION
**Pull request by tridge**
this allows beta version to be displayed

### CODE REVIEW

**Comment by rmackay9** [comment]
I hope that @meee1 can enhance MP so that the Firmware Install screen can show the Beta or RC version number.

This is what the screen displays at the moment and you can see there is no version number
image

**Comment by peterbarker** [comment]
Do we have before/after snippets for the generated manifest?

Which information present in this string wasn't available in other fields?

**Comment by peterbarker** [comment]
Would be nice to add the beta number to the data as a separate field so people aren't tempted to parse this new string.

I'd suggest we call this field "display-string", too.

File: Tools/scripts/generate_manifest.py [hunk 1]
```diff
@@ -163,6 +164,17 @@ def git_sha_from_git_version(self, filepath):
                 "filepath (%s) does not contain a git sha" % (filepath,))
         return m.group("sha")
 
+    def fwversion_from_git_version(self, filepath):
+        '''parses get-version.txt (as emitted by build_binaries.py, returns
+        git sha from it'''
+        content = open(filepath).read()
+        sha_regex = re.compile("APMVERSION: \S+\s+(\S+)")
```
Comment (magicrub) [inline Tools/scripts/generate_manifest.py:171]: What is this "APM" you speak of?

Comment (peterbarker) [reply Tools/scripts/generate_manifest.py:171]: :-)  It's present in vast numbers of pre-existing files on our build server - short of rewriting all those files we're kind of stuck with this ATM.  It was like that when I got here (pretty sure, anyway!)

We could consider understanding e.g. `ARDUPILOT_VERSION: ` as well and add another line into those files - but I don't know if there are external tools also looking for that string, soooo.....

**Comment by tridge** [comment]
> Would be nice to add the beta number to the data as a separate field so people aren't tempted to parse this new string.

we don't have that readily available, especially for existing builds

> I'd suggest we call this field "display-string", too.

mav-firmware-version-str matches the existing pattern, and "display-string" doesn't indicate it is a firmware version

File: Tools/scripts/generate_manifest.py (hunk 1 above)
Comment (magicrub) [reply Tools/scripts/generate_manifest.py:171]: >It was like that when I got here

LMAO

**Comment by magicrub** [comment]
@meee1 says he's already implemented it on MP side.

**Review by peterbarker — APPROVED** [review]

**Pull request merged**